### PR TITLE
Disables the CME event until it is reworked.

### PR DESCRIPTION
**Pull request by chemistrymain2**
## About The Pull Request

This PR turns the weight of the CME event to zero.

## How This Contributes To The Skyrat Roleplay Experience

Coronal Mass Ejections have a lot in common with TGstation's icebox earthquakes-they're random 'FUCK YOU' events that you can do nothing about and fuck over everyone and are very annoying. 

They pop up without warning.

Even when defused successfully, they still cause atmos damage-when not defused, they cause an EMP.

They are minorly aggravating and really only serve the same purpose as a radiation storm, but unlike an anomaly yields no reward for dealing with it.

You cannot tell where they're going to crop up before they crop up, either-no room shielding, nada. Sometimes it just shows up and fucks you for no reason.

It's the peak of 'this event annoys engineering', and it's probably for the best it be disabled until it can be appropriately reworked to something more interesting.

## Proof of Testing

image

## Changelog

:cl:
del: The Coronal Mass Ejection event will no longer fire.
/:cl:

### CODE REVIEW

**Comment by lessthnthree** [comment]
You should remove meteors too, they're in the same vein right?

**Comment by SciDragon** [comment]
Please, for the love of all that is holy, stop removing events from Skyrat.

**Comment by jjpark-kb** [comment]
sometimes events are left best to be a nuisance/annoyance rather than a full blown station event... I'm speaking from experience here (spacevines).

**Comment by TheGandalf85** [comment]
Some design musings in case anyone wants to work on a modification to it later on. These are only suggestions and my thoughts, and never a criticism on the developer or their work themselves.

While it's never the end of the world, be sure to be careful on removal of anything that gives jobs things to do, no matter how small. Of course, tweaking the chances instead of outright removal is also an option.

EMP is **pretty devastating**, especially at the scale the CME does it at. It also does it passively even while it's out. (Citation required, though I remember siphons being set despite disarming...) This shouldn't happen at bare minimum unless the CME pops. EMPs at this level fucks with air alarms, completely destroys any engineering set up just for existing, and can technically insta-gib certain mobs if extremely unlucky. The heating it also provides isn't a good thing since there's no way to automatically stabilize temperatures without dragging a space heater, but this is arguable due to their availability in maintenance tunnels.

My recommendation for the above is not let the resulting explosion from a pop destroy any tiles/walls, but allow an EMP to happen if it does pop. Challenging and fixable if it does go off. This keeps it in line with the "annoyance level" of other events like carp or bluespace reds.

No rewards is bad. Needs to drop raw anomalies at the very least. This turns it from an annoyance to a rewarding work time for engineering/science.

The length of time it takes to pop is okay. The station is (usually) given a warning and this gives ample time for others to prep.

There should be another way for everyone to be able to contribute to CMEs. Currently, only Science can handle it at all and if they're asleep, we're boned. Similar to Anomalies, they should be able to be popped with a signaler/analyzer combo.

**Comment by Zergspower** [comment]
Can we not do PR's 'until it's X' because it never gets done

**Comment by chemistrymain2** [comment]
> You should remove meteors too, they're in the same vein right?

You can counter meteors.

You cannot counter CME's.

Which is to say the event is inherently uncounterable because even if you're on the ball with neutralizers it still fucks atmos.

**Comment by Namwty** [comment]
> they're random 'FUCK YOU' events

Frankly this is so subjective that it can be said about any event/antag that ss13 has to offer, be it nukies/blob or whatever else. For example I don't think mold is bad, but there are people who think it's "FUCK YOU" event.

From general standpoint, antags ( yeah, if skyrat still has them left ) use these "FUCK YOU" events to make their moves. For example, It's easier to take someone hostage and prep a scene, when security or generally crew is busy with stuff ( example: mold ), than when everyone is so bored that even slightest antag's fart makes whole security team + department you involed in pile on you.

In short, chaos events gives room to antags to do stuff and sometimes even creates atmosphere that helps said scenes.

> Can we not do PR's 'until it's X' because it never gets done

Yeah, lately a lot of PRs, be it in SR or base TG have this mentality.
Even worse is when someone introduces something "half-baked" saying that it will be "tweaked/updated" with these fancy "TO-DO LISTS" that are meant to be finished, but never is once stuff is merged.

Regardless, Zerg is right.

**Comment by chemistrymain2** [comment]
> Please, for the love of all that is holy, stop removing events from Skyrat.

There is not one single person who enjoys dealing with this any more than they enjoyed dealing with wall fungus.

They share similar design philosophy failures-a lack of reward, a limited method of contribution, general uninteractiveness, extremely punishing, and effectively random.

Compare to meteors, for instance. Meteors can be countered with meteor shields. Likewise, meteors can be countered with pre-emptive defense-you can build walls to block them off, et cetera. Multiple departments can contribute to solving the problem, or by repairing the damage they cause. Botany, chemistry, et cetera can all contribute to meteors,  instead of just engineering, and multiple departments are involved in the resolution of them.

Likewise, you're also told the direction the meteors are coming from, get an alert, so on, so forth.

CME's, by comparison, are random. You get an alert that the station is going to get fucked. Then, for a various period of time, a random number of anomalies spawn around the station in randomly chosen spots.

Not only does this have problems such as unclear locations ('CME in 'space', CME in 'station asteroid') or hitting protected areas such as the dorms, but they're handleable by ONE department, and that's science.

No other department can deal with them, per se. There's no other interaction to them than 'hit them with an anomaly neutralizer'. Nothing any other department can contribute has a meaningful effect on the problem. Does cargo have CME shields they can order? No. Can botany meaningfully do anything about the problem? No. Can anyone without an anomaly neutralizer do anything? No. Can chemistry make some sort of reagent like they can a breach sealant to neutralize the effect? No. So on, so forth.

With kudzu, many departments can contribute. Even the chef can get in on the action, with their goat. With anomalies, any department can resolve the issue, and many anomalies involve different departments-bioscrambler anomalies involve medbay, for instance, and materials anomalies can be farmed for resources. The cores themselves are rewards, useful in science, which a lot of R&D involves.

CME's just kind of exist. They are a 'fuck you' because they have no interaction, unlike other events. They have _one_ involved department, with _one_ method of interaction. Unlike a rod, these hit a whole lot of places, unlike space carp or shocktroopers they're not something you can fight- you just kind of have to hope that the sun doesn't spawn on top of you.

And if they pop, they cause a lot of annoying damage to atmospherics with a plasma injection and fuck up the atmos in the room for like 20 minutes so you get to listen to the fire alarms whine, and they set off EMPs.

There, that's my wall of text reasoning for 'why this event is bad'. It's not at all like meteors or kudzu or anything of the sort- it's just 'HAHA GET TO IT PRINT NEUTRALIZERS OR BE DISRUPTED'. 

At least radstorm was thematic about it.

**Comment by Nerev4r** [comment]
> Contributor

ok 

> > Please, for the love of all that is holy, stop removing events from Skyrat.
> 
> There is not one single person who enjoys dealing with this any more than they enjoyed dealing with wall fungus.
> 
> They share similar design philosophy failures-a lack of reward, a limited method of contribution, general uninteractiveness, extremely punishing, and effectively random.
> 
> Compare to meteors, for instance. Meteors can be countered with meteor shields. Likewise, meteors can be countered with pre-emptive defense-you can build walls to block them off, et cetera. Multiple departments can contribute to solving the problem, or by repairing the damage they cause. Botany, chemistry, et cetera can all contribute to meteors, instead of just engineering, and multiple departments are involved in the resolution of them.
> 
> Likewise, you're also told the direction the meteors are coming from, get an alert, so on, so forth.
> 
> CME's, by comparison, are random. You get an alert that the station is going to get fucked. Then, for a various period of time, a random number of anomalies spawn around the station in randomly chosen spots.
> 
> Not only does this have problems such as unclear locations ('CME in 'space', CME in 'station asteroid') or hitting protected areas such as the dorms, but they're handleable by ONE department, and that's science.
> 
> No other department can deal with them, per se. There's no other interaction to them than 'hit them with an anomaly neutralizer'. Nothing any other department can contribute has a meaningful effect on the problem. Does cargo have CME shields they can order? No. Can botany meaningfully do anything about the problem? No. Can anyone without an anomaly neutralizer do anything? No. Can chemistry make some sort of reagent like they can a breach sealant to neutralize the effect? No. So on, so forth.
> 
> With kudzu, many departments can contribute. Even the chef can get in on the action, with their goat. With anomalies, any department can resolve the issue, and many anomalies involve different departments-bioscrambler anomalies involve medbay, for instance, and materials anomalies can be farmed for resources. The cores themselves are rewards, useful in science, which a lot of R&D involves.
> 
> CME's just kind of exist. They are a 'fuck you' because they have no interaction, unlike other events. They have _one_ involved department, with _one_ method of interaction. Unlike a rod, these hit a whole lot of places, unlike space carp or shocktroopers they're not something you can fight- you just kind of have to hope that the sun doesn't spawn on top of you.
> 
> And if they pop, they cause a lot of annoying damage to atmospherics with a plasma injection and fuck up the atmos in the room for like 20 minutes so you get to listen to the fire alarms whine, and they set off EMPs.
> 
> There, that's my wall of text reasoning for 'why this event is bad'. It's not at all like meteors or kudzu or anything of the sort- it's just 'HAHA GET TO IT PRINT NEUTRALIZERS OR BE DISRUPTED'.
> 
> At least radstorm was thematic about it.

ok you're kinda spittin with this one and that's true
there's no real... gameplay to it. there's nothing most departments can provide besides empty hands to hold neutralizers

**Comment by SciDragon** [comment]
> > Please, for the love of all that is holy, stop removing events from Skyrat.
> 
> There is not one single person who enjoys dealing with this any more than they enjoyed dealing with wall fungus.
> 
> They share similar design philosophy failures-a lack of reward, a limited method of contribution, general uninteractiveness, extremely punishing, and effectively random.
> 
> Compare to meteors, for instance. Meteors can be countered with meteor shields. Likewise, meteors can be countered with pre-emptive defense-you can build walls to block them off, et cetera. Multiple departments can contribute to solving the problem, or by repairing the damage they cause. Botany, chemistry, et cetera can all contribute to meteors, instead of just engineering, and multiple departments are involved in the resolution of them.
> 
> Likewise, you're also told the direction the meteors are coming from, get an alert, so on, so forth.
> 
> CME's, by comparison, are random. You get an alert that the station is going to get fucked. Then, for a various period of time, a random number of anomalies spawn around the station in randomly chosen spots.
> 
> Not only does this have problems such as unclear locations ('CME in 'space', CME in 'station asteroid') or hitting protected areas such as the dorms, but they're handleable by ONE department, and that's science.
> 
> No other department can deal with them, per se. There's no other interaction to them than 'hit them with an anomaly neutralizer'. Nothing any other department can contribute has a meaningful effect on the problem. Does cargo have CME shields they can order? No. Can botany meaningfully do anything about the problem? No. Can anyone without an anomaly neutralizer do anything? No. Can chemistry make some sort of reagent like they can a breach sealant to neutralize the effect? No. So on, so forth.
> 
> With kudzu, many departments can contribute. Even the chef can get in on the action, with their goat. With anomalies, any department can resolve the issue, and many anomalies involve different departments-bioscrambler anomalies involve medbay, for instance, and materials anomalies can be farmed for resources. The cores themselves are rewards, useful in science, which a lot of R&D involves.
> 
> CME's just kind of exist. They are a 'fuck you' because they have no interaction, unlike other events. They have _one_ involved department, with _one_ method of interaction. Unlike a rod, these hit a whole lot of places, unlike space carp or shocktroopers they're not something you can fight- you just kind of have to hope that the sun doesn't spawn on top of you.
> 
> And if they pop, they cause a lot of annoying damage to atmospherics with a plasma injection and fuck up the atmos in the room for like 20 minutes so you get to listen to the fire alarms whine, and they set off EMPs.
> 
> There, that's my wall of text reasoning for 'why this event is bad'. It's not at all like meteors or kudzu or anything of the sort- it's just 'HAHA GET TO IT PRINT NEUTRALIZERS OR BE DISRUPTED'.
> 
> At least radstorm was thematic about it.

I can assure you that engineers like doing engineering and atmospherics work, which this event facilitates. There are plenty of situations that are intended to be dealt with by only one or two departments, it's not like we're calling on service to deal with contractors. When meteors or rods DO happen, do you know which department is called to fix them? Engineering. Service doesn't do it. Science doesn't do it. Security doesn't do it. 
Why is it not okay for events like this to just exist and be annoying? Why does everything that could be considered a nuisance have to be systematically removed until we have nothing left? If you want the CMEs to have gameplay, then yes, I agree with Gandalf in some part, let them be destroyed by the analyzer/signaller combo.

**Comment by lessthnthree** [comment]
What does immovable rod gameplay bring, then.

**Comment by Hatterhat** [comment]
> What does immovable rod gameplay bring, then.

schadenfreude in the event that it hits someone, double schadenfreude if it's a looper and/or punches holes in *several* people

**Comment by chemistrymain2** [comment]
> What does immovable rod gameplay bring, then.

funny when it hits someone and kills them without being stretched out for too long + opportunity for RD to do something

look not all of the events are winners but immovable rod has been in the game for longer than cme's

**Comment by chemistrymain2** [comment]
> > > Please, for the love of all that is holy, stop removing events from Skyrat.
> > 
> > 
> > There is not one single person who enjoys dealing with this any more than they enjoyed dealing with wall fungus.
> > They share similar design philosophy failures-a lack of reward, a limited method of contribution, general uninteractiveness, extremely punishing, and effectively random.
> > Compare to meteors, for instance. Meteors can be countered with meteor shields. Likewise, meteors can be countered with pre-emptive defense-you can build walls to block them off, et cetera. Multiple departments can contribute to solving the problem, or by repairing the damage they cause. Botany, chemistry, et cetera can all contribute to meteors, instead of just engineering, and multiple departments are involved in the resolution of them.
> > Likewise, you're also told the direction the meteors are coming from, get an alert, so on, so forth.
> > CME's, by comparison, are random. You get an alert that the station is going to get fucked. Then, for a various period of time, a random number of anomalies spawn around the station in randomly chosen spots.
> > Not only does this have problems such as unclear locations ('CME in 'space', CME in 'station asteroid') or hitting protected areas such as the dorms, but they're handleable by ONE department, and that's science.
> > No other department can deal with them, per se. There's no other interaction to them than 'hit them with an anomaly neutralizer'. Nothing any other department can contribute has a meaningful effect on the problem. Does cargo have CME shields they can order? No. Can botany meaningfully do anything about the problem? No. Can anyone without an anomaly neutralizer do anything? No. Can chemistry make some sort of reagent like they can a breach sealant to neutralize the effect? No. So on, so forth.
> > With kudzu, many departments can contribute. Even the chef can get in on the action, with their goat. With anomalies, any department can resolve the issue, and many anomalies involve different departments-bioscrambler anomalies involve medbay, for instance, and materials anomalies can be farmed for resources. The cores themselves are rewards, useful in science, which a lot of R&D involves.
> > CME's just kind of exist. They are a 'fuck you' because they have no interaction, unlike other events. They have _one_ involved department, with _one_ method of interaction. Unlike a rod, these hit a whole lot of places, unlike space carp or shocktroopers they're not something you can fight- you just kind of have to hope that the sun doesn't spawn on top of you.
> > And if they pop, they cause a lot of annoying damage to atmospherics with a plasma injection and fuck up the atmos in the room for like 20 minutes so you get to listen to the fire alarms whine, and they set off EMPs.
> > There, that's my wall of text reasoning for 'why this event is bad'. It's not at all like meteors or kudzu or anything of the sort- it's just 'HAHA GET TO IT PRINT NEUTRALIZERS OR BE DISRUPTED'.
> > At least radstorm was thematic about it.
> 
> I can assure you that engineers like doing engineering and atmospherics work, which this event facilitates. There are plenty of situations that are intended to be dealt with by only one or two departments, it's not like we're calling on service to deal with contractors. When meteors or rods DO happen, do you know which department is called to fix them? Engineering. Service doesn't do it. Science doesn't do it. Security doesn't do it. Why is it not okay for events like this to just exist and be annoying? Why does everything that could be considered a nuisance have to be systematically removed until we have nothing left? If you want the CMEs to have gameplay, then yes, I agree with Gandalf in some part, let them be destroyed by the analyzer/signaller combo.

You know what I realized I don't care to argue this further.

They suck.

I'd like to axe them because nobody enjoys the event. It's not worth keeping something that people don't enjoy because 'but we don't have anything else!' There's plenty of other things we do have.

This one can go to hell and you have my wallpost on the matter. I won't argue further.

**Comment by SpaceLoveSs13** [comment]
This is a good solely because I can tell as an engineering player and echo other engineering players, fixing CMEs, fixing air is not fun. I don't know what others are talking about with "facilitating engineers to do engineering things". It simply blocks actual engineers and Atmos techs from doing their stuff.

**Comment by Stalkeros2** [comment]
Just delete them entirely; Cobalt's pretty much gone so nobody's gonna work on them.

**Comment by ghost** [comment]
For the longest time I said "I'll rework them to be like real life CMEs one day" since it HURTS to see a neat science thing made into "lol death lasers"

So yeah 👍 from me this event isn't fun and "but meteors but rod" aren't excuses. This was a modular addition that does not provide joy. All it provides are siphoned air alarms. Which suck. A lot.

**Comment by desertmote** [comment]
CMEs give _some_ sort of warning. How about an energy draining machine you can put in an area, once its installed and enabled it will suppress CMEs that happen to appear in this area?

**Comment by Plum-but-gayer** [comment]
can we bring back rad storms from bay? I miss shuffling into maint with a bunch of randos and smoking a cig while we rp for 5 minutes

anyway fuck CMEs lmao. If plasma atmos and fire alarms were less jank and terrible I could be convinced they could stay around but tbh they just create sooo much unfun work for engies.

**Comment by HoodMilk** [comment]
Crazy how nobody has mentioned vortex anomalies, which exist to instantly destroy and space a room with no warning or way to counter it before the damage is done.

**Comment by chemistrymain2** [comment]
> CMEs give _some_ sort of warning. How about an energy draining machine you can put in an area, once its installed and enabled it will suppress CMEs that happen to appear in this area?

This, which I don't have sprites for, would resolve one of like eight issues with the event. I point to icebox earthquakes, which are basically the same thing-it gives you a few minutes of warning that your department is going to explode and then your department explodes, and you have to get a machine from cargo to protect a random area and stop your department from exploding (it only works if it is near the area that will explode.)

That's still bad gameplay.

Fixing this event would involve overhauling basically everything. The method of defusal. The way other departments can interact with it. The rewards for doing so. The relative utility of it. How it responds when defused. By the end of it, i'd have an entirely different event.

The BEST we can come up with for this is 'eight pyro anomalies spawn in random spots' and our current best solution to it is 'make defusing these pyro anomalies not kill the person who did it, make it doable with an analyzer, and maybe uhh an anomaly blocker machine.'

I'm not interested in that kind of resolution because that kind of resolution sucks ass. Even if you polish a turd it's still a turd. The event needs a comprehensive overhaul from the ground up, not just 'at least now you can defuse it with more than just one thing'.

And i'm not going to do that comprehensive overhaul because honestly dude I don't want to I have better things to do with my time and there are better events that should be rescued from the scrapheap (blob, radstorm, etc)

**Comment by TheGandalf85** [comment]
> This is a good solely because I can tell as an engineering player and echo other engineering players, fixing CMEs, fixing air is not fun. I don't know what others are talking about with "facilitating engineers to do engineering things". It simply blocks actual engineers and Atmos techs from doing their stuff.

Usually going on a job slot while not doing the job is poor form. If you go Engineer/Atmos Tech and not expect to repair the station, it's bad form. It actually does suck when you want to go on full 3 hour project monkey mode, but their literal primary purpose is to keep the station as repaired and in-working-order as possible. Fixing CME's might be objectively unfun, but you can't say fixing air isn't, especially for everyone who runs the atmos tech slot.

**Comment by TheGandalf85** [comment]
> Crazy how nobody has mentioned vortex anomalies, which exist to instantly destroy and space a room with no warning or way to counter it before the damage is done.

Yep. This anomaly is absolutely shnasty. Instant, uncounterable, hole on the floor and potentially multiple rooms. I've been RRd by it multiple times as an engineer, and only did I realize recently you can throw an Anomaly Neutralizer at it. I personally think this is a different conversation however. This gives engineers something to do. It's mostly just how destructive it is especially to certain circumstances like it appearing anywhere near Dorm room 4.

**Comment by TheGandalf85** [comment]
The primary issue I'm seeing here with everyone's sort of look into this is that the CME isn't engaging enough. I don't think the disruption is the issue here. It's just the fact that there's absolutely 0 reward, and when it happens it's just a "ah fuck here we go again" moment. You can make something that inhibits the round and gives people things they have to work on or suffer, but also make it actually engaging. Anomalies suck, but they give a very nice resource. For science it's basically a free gift basket, and I never have heard a Scientist not dislike anomalies despite their nature.

As for destruction, simply scale events based on their destructive nature. We're an MRP/HRP server, so these sort of events should be rare. CME's are counterable and it's possible to have 0 reprocussions if science is on their a game, so I just believe the thing needs some rewards on it, along with other people being able to contribute to the event.

**Comment by burgerenergy** [comment]
What rewards are necessary? After a hour the station has infinite resources, CMEs add a bit of spice to the round in the form of a physical difficulty that can be worked around and, of course, fixed. This is particularly important given we, in the royal we of Skyrat, have basically stomped down any possibility of mass destruction via players, so events are basically the only way the station proper receives damage. 

Is the event tedious? Of course it is after the first couple of times, but you're playing a stagnate, ten year old shitty that you can see all the mechanical content within a week of casual play. You're bored, but making things more stagnate is not the answer.

**Comment by chemistrymain2** [comment]
> What rewards are necessary? After a hour the station has infinite resources, CMEs add a bit of spice to the round in the form of a physical difficulty that can be worked around and, of course, fixed. This is particularly important given we, in the royal we of Skyrat, have basically stomped down any possibility of mass destruction via players, so events are basically the only way the station proper receives damage.
> 
> Is the event tedious? Of course it is after the first couple of times, but you're playing a stagnate, ten year old shitty that you can see all the mechanical content within a week of casual play. You're bored, but making things more stagnate is not the answer.

It's tedious like wall fungus was tedious. Bad design is its death knell.

**Comment by burgerenergy** [comment]
No it isn't. Wall fungus was tedious because you had to a.) hunt it down at an unannounced point on the station b.) typically had to destroy walls and go into space to get it all c.) the visceral gameplay of clicking walls with a welder.

This is, quite literally, a flash fire. It causes engineering work, some injuries if careless, and disrupts the station in ultimately manageable ways. If you think this is unplayably tedious then put up a PR to remove random diseases, the unmoveable rod, and carp.

**Comment by chemistrymain2** [comment]
this can either be closed or merged but i'm tired of clicking 'update'

**Comment by Gandalf2k15** [comment]
Removing events is not the way we want to be going anymore.
